packages/data-context/src/sources/HtmlDataSource.ts

```diff
@@ -75,6 +75,16 @@ export class HtmlDataSource {
       // Error getting config, we will show an error screen when we render the page
     }
 
+    // for project-base config, the remote state we wish to convey should be whatever top is set to, also known as the primary domain
+    // whenever the app is served/re-served
+    if (this.ctx.coreData.currentTestingType === 'e2e') {
```

Comment:
From what it looks like, `makeServeConfig` in `data-context -> HtmlDataSource` is now accomplishing what `runner.serve` use to accomplish in the [original](https://github.com/cypress-io/cypress/pull/18075/files#diff-7ec134b05d5f4b3f8dac0940baefe0ecd2b6bd331acf11ab4332cfe87fec8096R71) implementation. @mschile and I both noticed that it doesn't look like `runner.serve` is used any longer in the `10.0-release` branch. Is this the case and are we missing something?

Reply:
Do we also need/want to do this for component testing?